helix-core/src/main/java/org/apache/helix/manager/zk/DistributedLeaderElection.java

```diff
@@ -87,36 +88,39 @@ public synchronized void onControllerChange(NotificationContext changeContext) {
 
   private void relinquishLeadership(HelixManager manager,
       ControllerManagerHelper controllerHelper) {
-    LOG.info(manager.getInstanceName() + " relinquish leadership for cluster: " + manager
+    long start = System.currentTimeMillis();
+    LOG.info(manager.getInstanceName() + " tries to relinquish leadership for cluster: " + manager
         .getClusterName());
     controllerHelper.stopControllerTimerTasks();
     controllerHelper.removeListenersFromController(_controller);
     // clear write-through cache
     manager.getHelixDataAccessor().getBaseDataAccessor().reset();
+    LOG.info("{} relinquishes leadership for cluster: {}, took: {}", manager.getInstanceName(),
```

Comment:
Correct me if I am wrong, I think we record every "onControllerChange" processing time. So it should be easy to derive the latency from that log. I feel this log would be redundant a little bit.

Reply:
It just makes the debugging process a little easier. Plus, it only adds the time spent on the method only in addition to the existing log. The "onControllerChange" process time can be the additional weapon to figure out the remaining latency other than the method.